### PR TITLE
Add support for UUIDs

### LINKED ISSUE #1503: [elm] Question is UUID referenced but not generated?
I'm not sure about this issue, so I'll raise it in form of a question.

The elm client generation creates data types like this (shortened).

```elm

module Data.Label exposing (Label, Type(..), decoder, encoder)

import Data.UUID as UUID exposing (UUID)
import Dict exposing (Dict)
import Json.Decode as Decode exposing (Decoder)
import Json.Decode.Pipeline exposing (optional, required)
import Json.Encode as Encode


type alias Label =
    { id : Maybe UUID
    , type_ : Type
    , text : String
    }

```

but there is no `Data.UUID` file.

There is also no dependency in the `elm.json` template:
https://github.com/OpenAPITools/openapi-generator/blob/b24b6df448878cd8eccae234644e5123785295b5/modules/openapi-generator/src/main/resources/elm/elm.mustache

#### Fix

Added the dependency `"danyx23/elm-uuid": "2.1.2"` and a `Data/UUID.elm` file.

```elm
module Data.UUID exposing (UUID, decoder, encoder)

import Json.Decode as Decode exposing (Decoder)
import Json.Encode as Encode
import Uuid


type alias UUID =
    Uuid.Uuid


decoder : Decoder UUID
decoder =
    Uuid.decoder


encoder : UUID -> Encode.Value
encoder =
    Uuid.encode
```

Am I missing something or is the client generation missing something?

### PR DESCRIPTION
**Pull request by trenneman**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.4.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Add support for UUIDs via [danyx23/elm-uuid](https://package.elm-lang.org/packages/danyx23/elm-uuid/latest). As this package (correctly) uses `encode` for encoding values (instead of `encoder`), this PR also fixes the naming for `decode` functions. When you do not use the `decoder` functions yourself you do not needs to update any code.

Fix #1503. cc @wing328

### CODE REVIEW

File: samples/client/petstore/elm-0.18/.openapi-generator/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-3.3.2-SNAPSHOT
\ No newline at end of file
+3.3.4-SNAPSHOT
```
Comment (andys8) [inline samples/client/petstore/elm-0.18/.openapi-generator/VERSION]: The renaming of `encode` will be a breaking change for me (no problem though). Don't know about versioning in this project, but if this is the version of the whole openapi generator-project, then it's probably fine :)

**Comment by andys8** [comment]
Looking great. Thanks for the changes :)

File: samples/client/petstore/elm-0.18/.openapi-generator/VERSION (hunk 1 above)
Comment (trenneman) [reply samples/client/petstore/elm-0.18/.openapi-generator/VERSION]: 3.3.4 will be the next release, hence the change.

Comment (wing328) [reply samples/client/petstore/elm-0.18/.openapi-generator/VERSION]: Due to the renaming of `encode`, I would suggest targeting this change for 4.0.x branch (v4.0.0 will be released in Dec 20th)

Comment (trenneman) [reply samples/client/petstore/elm-0.18/.openapi-generator/VERSION]: Typically you do not need to use the `encode` directly. But if this is considered a 'Breaking change (without fallback)' I'll target it for 4.0.x.

**Pull request merged**